cpp/src/arrow/util/formatting_util_test.cc

```diff
@@ -522,6 +522,28 @@ TEST(Formatting, Timestamp) {
     AssertFormatting(formatter, -2203932304LL * 1000000000LL + 8,
                      "1900-02-28 12:34:56.000000008");
   }
+
+  {
+    auto timestamp_types = {timestamp(TimeUnit::SECOND, "US/Eastern"),
+                            timestamp(TimeUnit::SECOND, "+01:00"),
+                            timestamp(TimeUnit::SECOND, "-42:00"),
+                            timestamp(TimeUnit::SECOND, "Pacific/Maruesas"),
+                            timestamp(TimeUnit::SECOND, "Mars/Mariner_Valley")};
```

Comment:
Can you also test with different resolutions?

(it might not be needed to test the whole list of values below for each type, we maybe can also just test one of them per resolution/tz outside of a loop, if that makes things simpler)